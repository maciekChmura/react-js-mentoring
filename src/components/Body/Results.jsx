import React, { Component } from 'react';
import styled from 'styled-components';
import MovieTile from './MovieTile';

const ResultsWrapper = styled.div`
  grid-area: 3 / 2 / 4 / 3;
  min-height: 500px;
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
`;

class Results extends Component {
  render() {
    return (
      <ResultsWrapper>
        {this.props.results.map(result => (
          <MovieTile key={result.id} data={result} />
        ))}
      </ResultsWrapper>
    );
  }
}

export default Results;
