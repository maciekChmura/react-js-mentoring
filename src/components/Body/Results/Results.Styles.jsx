import styled from 'styled-components';

const ResultsWrapper = styled.div`
  grid-area: 3 / 2 / 4 / 3;
  min-height: 500px;
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
  justify-content: space-around;
  align-content: flex-start;
`;

export default ResultsWrapper;
