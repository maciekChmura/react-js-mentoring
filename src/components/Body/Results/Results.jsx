import React from 'react';
import ResultsWrapper from './Results.Styles';
import MovieTile from '../MovieTile/MovieTile';

const Results = ({ results, changePage }) => (
  <ResultsWrapper>
    {results.map(result => (
      <MovieTile key={result.id} data={result} changePage={changePage} />
    ))}
  </ResultsWrapper>
);

export default Results;
