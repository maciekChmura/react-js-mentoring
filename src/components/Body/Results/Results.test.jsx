import React from 'react';
import { shallow } from 'enzyme';

import Results from './Results';

describe('Results', () => {
  const results = [
    {
      id: 68721,
      title: 'Iron Man 3',
      tagline: 'Unleash the power behind the armor.',
      vote_average: 6.8,
      vote_count: 10461,
      release_date: '2013-04-18',
      poster_path:
        'https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg',
      overview:
        "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
      budget: 200000000,
      revenue: 1215439994,
      genres: ['Action', 'Adventure', 'Science Fiction'],
      runtime: 130
    },
    {
      id: 1726,
      title: 'Iron Man',
      tagline: "Heroes aren't born. They're built.",
      vote_average: 7.4,
      vote_count: 10926,
      release_date: '2008-04-30',
      poster_path:
        'https://image.tmdb.org/t/p/w500/848chlIWVT41VtAAgyh9bWymAYb.jpg',
      overview:
        'After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.',
      budget: 140000000,
      revenue: 585174222,
      genres: ['Action', 'Science Fiction', 'Adventure'],
      runtime: 126
    }
  ];
  it('should render correctly', () => {
    const component = shallow(<Results results={results} />);

    expect(component).toMatchSnapshot();
  });
});
