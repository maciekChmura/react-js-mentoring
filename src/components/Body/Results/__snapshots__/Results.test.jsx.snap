// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    results={
      Array [
        Object {
          "budget": 200000000,
          "genres": Array [
            "Action",
            "Adventure",
            "Science Fiction",
          ],
          "id": 68721,
          "overview": "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
          "poster_path": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
          "release_date": "2013-04-18",
          "revenue": 1215439994,
          "runtime": 130,
          "tagline": "Unleash the power behind the armor.",
          "title": "Iron Man 3",
          "vote_average": 6.8,
          "vote_count": 10461,
        },
        Object {
          "budget": 140000000,
          "genres": Array [
            "Action",
            "Science Fiction",
            "Adventure",
          ],
          "id": 1726,
          "overview": "After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.",
          "poster_path": "https://image.tmdb.org/t/p/w500/848chlIWVT41VtAAgyh9bWymAYb.jpg",
          "release_date": "2008-04-30",
          "revenue": 585174222,
          "runtime": 126,
          "tagline": "Heroes aren't born. They're built.",
          "title": "Iron Man",
          "vote_average": 7.4,
          "vote_count": 10926,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MovieTile
          data={
            Object {
              "budget": 200000000,
              "genres": Array [
                "Action",
                "Adventure",
                "Science Fiction",
              ],
              "id": 68721,
              "overview": "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
              "poster_path": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
              "release_date": "2013-04-18",
              "revenue": 1215439994,
              "runtime": 130,
              "tagline": "Unleash the power behind the armor.",
              "title": "Iron Man 3",
              "vote_average": 6.8,
              "vote_count": 10461,
            }
          }
        />,
        <MovieTile
          data={
            Object {
              "budget": 140000000,
              "genres": Array [
                "Action",
                "Science Fiction",
                "Adventure",
              ],
              "id": 1726,
              "overview": "After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.",
              "poster_path": "https://image.tmdb.org/t/p/w500/848chlIWVT41VtAAgyh9bWymAYb.jpg",
              "release_date": "2008-04-30",
              "revenue": 585174222,
              "runtime": 126,
              "tagline": "Heroes aren't born. They're built.",
              "title": "Iron Man",
              "vote_average": 7.4,
              "vote_count": 10926,
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "68721",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "budget": 200000000,
            "genres": Array [
              "Action",
              "Adventure",
              "Science Fiction",
            ],
            "id": 68721,
            "overview": "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
            "poster_path": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
            "release_date": "2013-04-18",
            "revenue": 1215439994,
            "runtime": 130,
            "tagline": "Unleash the power behind the armor.",
            "title": "Iron Man 3",
            "vote_average": 6.8,
            "vote_count": 10461,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1726",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "budget": 140000000,
            "genres": Array [
              "Action",
              "Science Fiction",
              "Adventure",
            ],
            "id": 1726,
            "overview": "After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.",
            "poster_path": "https://image.tmdb.org/t/p/w500/848chlIWVT41VtAAgyh9bWymAYb.jpg",
            "release_date": "2008-04-30",
            "revenue": 585174222,
            "runtime": 126,
            "tagline": "Heroes aren't born. They're built.",
            "title": "Iron Man",
            "vote_average": 7.4,
            "vote_count": 10926,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ResultsStyles__ResultsWrapper-sc-16qle82-0",
        "isStatic": true,
        "rules": Array [
          "grid-area:3 / 2 / 4 / 3;min-height:500px;display:flex;flex-wrap:wrap;align-content:flex-start;justify-content:space-around;align-content:flex-start;",
        ],
      },
      "displayName": "ResultsStyles__ResultsWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "ResultsStyles__ResultsWrapper-sc-16qle82-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MovieTile
            data={
              Object {
                "budget": 200000000,
                "genres": Array [
                  "Action",
                  "Adventure",
                  "Science Fiction",
                ],
                "id": 68721,
                "overview": "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
                "poster_path": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
                "release_date": "2013-04-18",
                "revenue": 1215439994,
                "runtime": 130,
                "tagline": "Unleash the power behind the armor.",
                "title": "Iron Man 3",
                "vote_average": 6.8,
                "vote_count": 10461,
              }
            }
          />,
          <MovieTile
            data={
              Object {
                "budget": 140000000,
                "genres": Array [
                  "Action",
                  "Science Fiction",
                  "Adventure",
                ],
                "id": 1726,
                "overview": "After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.",
                "poster_path": "https://image.tmdb.org/t/p/w500/848chlIWVT41VtAAgyh9bWymAYb.jpg",
                "release_date": "2008-04-30",
                "revenue": 585174222,
                "runtime": 126,
                "tagline": "Heroes aren't born. They're built.",
                "title": "Iron Man",
                "vote_average": 7.4,
                "vote_count": 10926,
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "68721",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "budget": 200000000,
              "genres": Array [
                "Action",
                "Adventure",
                "Science Fiction",
              ],
              "id": 68721,
              "overview": "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
              "poster_path": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
              "release_date": "2013-04-18",
              "revenue": 1215439994,
              "runtime": 130,
              "tagline": "Unleash the power behind the armor.",
              "title": "Iron Man 3",
              "vote_average": 6.8,
              "vote_count": 10461,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1726",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "budget": 140000000,
              "genres": Array [
                "Action",
                "Science Fiction",
                "Adventure",
              ],
              "id": 1726,
              "overview": "After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.",
              "poster_path": "https://image.tmdb.org/t/p/w500/848chlIWVT41VtAAgyh9bWymAYb.jpg",
              "release_date": "2008-04-30",
              "revenue": 585174222,
              "runtime": 126,
              "tagline": "Heroes aren't born. They're built.",
              "title": "Iron Man",
              "vote_average": 7.4,
              "vote_count": 10926,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ResultsStyles__ResultsWrapper-sc-16qle82-0",
          "isStatic": true,
          "rules": Array [
            "grid-area:3 / 2 / 4 / 3;min-height:500px;display:flex;flex-wrap:wrap;align-content:flex-start;justify-content:space-around;align-content:flex-start;",
          ],
        },
        "displayName": "ResultsStyles__ResultsWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ResultsStyles__ResultsWrapper-sc-16qle82-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
